visualization/app/codeCharta/util/fileValidator.spec.ts

```diff
@@ -173,6 +173,28 @@ describe("FileValidator", () => {
 		}, expectedError)
 	})
 
+	describe("fixed sub folders validation", () => {
+		
+		it("should throw an error, if two sub folders horizontally overlap", () => {
+			file = clone(fileWithFixedOverlappingSubFolders)
+			const folder1: CodeMapNode = file.nodes[0].children[0].children[0]
+			const folder2: CodeMapNode= file.nodes[0].children[0].children[1]
+
+			const expectedError: CCValidationResult = {
+				error: [
+					`${ERROR_MESSAGES.fixedFoldersOverlapped} Found: folder_1_1 ${JSON.stringify(
+						folder1.fixedPosition
+					)} and folder_1_2 ${JSON.stringify(folder2.fixedPosition)}`
+				],
+				warning: []
+			}
+
+			assert.throws(() => {
+				validate(file)
+			}, expectedError)
+		})
+	})
+
```

Comment:
Maybe add a test that a sub-folder cant 'glitch' out of the main folder, as this kept happening during development?

Reply:
We have added tests like you describe in `treeMapGenerator.spec.ts` to ensure that fixed sub-folders will not be placed outside of their parent. Did you mean tests like this?